### ISSUE TITLE
How should --library compilation react to --no-local?

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#12684]
Low priority.  Logging so I don't forget about it.

How should `--library` compilation react to `--no-local`?  Should it create the library/executable pair like multi-locale interoperability will?  Or should it stay similar to single-locale compilation, but with the changes typical of a `--no-local` executable?

We probably should throw an error for `--no-local` + `--library` until this is decided, but I suspect doing so is also low-priority

**Comment by bradcray** [comment]
Are you asking specifically here about `--library` + `--no-local` + `--comm=none`?  I.e., I think of `--library` + `--no-local` as being the way to opt into what you're calling multi-locale interoperability (but also with `--comm=<non-none>`).

**Comment by lydia-duncan** [comment]
Yeah, basically.  You said something about this before, right?

**Comment by lydia-duncan** [comment]
Logically I view this as an extension, since the main goal is to get `--library` to work for multi-locale compilation.  My hope is that triggering it for `--no-local --comm=none` once support is present for `CHPL_COMM != none` will be simpler than adding the support in the first place, and I think the work will go faster if we do it in that order, rather than making the extension to support both a gating factor to the work.

That said, I could very well be missing key information, so it's okay if I'm wrong here

**Comment by bradcray** [comment]
I agree that comm != none is the important case to focus on in the short-term.  I was just trying to clarify the issue for myself since it doesn't say anything about the comm setting (so my first reading of it was confused since comm != none results in `--no-local` being thrown by default.  So the main question — "How should --library compilation react to --no-local?" — surprised me since that seemed like what we've been working on for awhile now until I realized you were probably talking specifically about the comm=none case).

**Comment by lydia-duncan** [comment]
Ah, that's fair.  Sorry to be confusing :)
